src/ucp/wireup/wireup.c

```diff
@@ -1527,6 +1529,21 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, unsigned ep_init_flags,
     ucp_ep_config_key_init_flags(&key, ep_init_flags);
     ucp_wireup_eps_pending_extract(ep, &replay_pending_queue);
 
+    /* Allow to choose only the lanes that were already chosen for case
+     * without CM to prevent reconfiguration error.
+     */
+    if ((ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) &&
+        !ucp_ep_has_cm_lane(ep)) {
+        UCS_BITMAP_CLEAR(&current_tl_bitmap);
+        for (lane = 0; lane < ucp_ep_config(ep)->key.num_lanes; ++lane) {
+            rsc_idx = ucp_ep_config(ep)->key.lanes[lane].rsc_index;
+
+            UCS_BITMAP_SET(current_tl_bitmap, rsc_idx);
+            ucs_assert(UCS_BITMAP_GET(tl_bitmap, rsc_idx));
```

Comment:
maybe ucs_assertv ?to print the missing resource name/index

Reply:
done